bob-backend/src/pearl/group.rs

```diff
@@ -517,6 +517,27 @@ impl Group {
         Ok(removed)
     }
 
+    pub async fn detach_by_id(&self, id: &str) -> BackendResult<Holder> {
+        let mut holders = self.holders.write().await;
+        debug!("write lock acquired");
+        let ts = get_current_timestamp();
+        for ind in 0..holders.len() {
+            if let Some(holder) = holders.get_child(ind) {
+                if holder.data.get_id() == id {
```

Comment:
`get_id()` allocates a new `String` on every call. Please, change its result type to `&str`. It is better to get `&str` and call `to_owned()` only in places where it is actually needed